main.ml

```diff
@@ -568,29 +568,47 @@ let process root_dir name =
       let open Deferred.Result.Monad_infix in
       fork_exec_rpc root_dir (script root_dir name `Volume "SR.probe") args Xapi_storage.Volume.Types.SR.Probe.Out.t_of_rpc
       >>= fun response ->
-      let srs = List.filter_map ~f:(fun probe_result ->
-        match probe_result.sr with
-        | Some sr_stat -> Some (sr_stat.Xapi_storage.Volume.Types.sr, {
-          Storage_interface.name_label = sr_stat.Xapi_storage.Volume.Types.name;
-        name_description = sr_stat.Xapi_storage.Volume.Types.description;
-        total_space = sr_stat.Xapi_storage.Volume.Types.total_space;
-        free_space = sr_stat.Xapi_storage.Volume.Types.free_space;
-        clustered = sr_stat.Xapi_storage.Volume.Types.clustered;
-        health = match sr_stat.Xapi_storage.Volume.Types.health with
-          | Xapi_storage.Volume.Types.Healthy _ -> Healthy
-          | Xapi_storage.Volume.Types.Recovering _ -> Recovering
-        });
-        | None -> None
-      ) response in
-      let uris = List.filter_map ~f:(fun probe_result ->
-        (* FIXME *)
-        if probe_result.complete then
-          Some (Jsonrpc.to_string (R.Dict (List.Assoc.map ~f:R.rpc_of_string probe_result.configuration)))
-        else
-          None
-      ) response in
-      let result = Storage_interface.(Probe { srs; uris }) in
-      Deferred.Result.return (R.success (Args.SR.Probe.rpc_of_response result))
+        let pp_probe_result () probe_result =
+          Xapi_storage.Volume.Types.SR.Probe.Out.rpc_of_t [probe_result] |>
+          Jsonrpc.to_string
+        in
+        response
+        |> List.map ~f:(fun probe_result ->
+          let uuid = List.Assoc.find probe_result.Xapi_storage.Volume.Types.configuration ~equal:String.equal "uuid" in
+          let open Deferred.Or_error in
+          match probe_result.Xapi_storage.Volume.Types.sr, probe_result.Xapi_storage.Volume.Types.complete, uuid with
+          | Some sr_stat, false, Some _uuid ->
+              return (`Attachable (probe_result.configuration, {
+                  Storage_interface.name_label = sr_stat.Xapi_storage.Volume.Types.name;
+                  name_description = sr_stat.Xapi_storage.Volume.Types.description;
+                  total_space = sr_stat.Xapi_storage.Volume.Types.total_space;
+                  free_space = sr_stat.Xapi_storage.Volume.Types.free_space;
+                  clustered = sr_stat.Xapi_storage.Volume.Types.clustered;
+                  health = match sr_stat.Xapi_storage.Volume.Types.health with
+                  | Xapi_storage.Volume.Types.Healthy _ -> Healthy
+                  | Xapi_storage.Volume.Types.Recovering _ -> Recovering
+                }));
+          | Some _sr, true, _ ->
+              errorf "A configuration cannot be suitable for both SR.create and SR.attach: %a" pp_probe_result probe_result
```

Comment:
I assume this is because the existence of the `sr` field means that there already exists an attached SR with this configuration, which means that we cannot create one with the same configuration?

Reply:
Yes, you would first have to destroy that SR.
If you are on an NFS(ng) SR then you can create multiple SRs in the same location as long as you use different UUIDs. So for NFS the plugin should list a location twice: once where it found the SR together with the UUID, and once more without the UUID meant for SR.create.